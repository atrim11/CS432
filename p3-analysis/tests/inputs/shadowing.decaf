int a;

def int main() {
    int i[2];
    void a[30];
    i = 0;

    while (i < 100) {
        bool a;

        if (i == 50) {
            int a[9999999];
        }
        i = i + 1;
    }
    return 1;
}
