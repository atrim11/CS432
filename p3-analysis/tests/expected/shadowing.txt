Void variable 'a' on line 5
Array 'i' accessed without index on line 6
Array 'i' accessed without index on line 8
Array 'i' accessed without index on line 11
Local variable 'a' on line 12 cannot be an array
Array 'i' accessed without index on line 14
Array 'i' accessed without index on line 14
Local variable 'i' on line 4 cannot be an array
Local variable 'a' on line 5 cannot be an array
